### PR TITLE
Convert float32 pi/2 value to float64 pi/2 value in SimTelEventSource

### PR DESCRIPTION
**Pull request by maxnoe**

### CODE REVIEW

File: ctapipe/io/simteleventsource.py [hunk 1]
```diff
@@ -878,9 +903,12 @@ def _fill_simulated_event_information(array_event):
         if mc_shower is None:
             return
 
+        # rounding errors might lead to values slightly above/below PI/2
```
Comment (nbiederbeck) [inline ctapipe/io/simteleventsource.py:906]: above/below +/- PI/2

File: ctapipe/io/simteleventsource.py [hunk 2]
```diff
@@ -73,6 +73,27 @@
...
+def _clip_altitude_if_close(altitude):
+    """
+    Round values slightly larger than pi/2 in float64 to pi/2
+
+    These can come from simtel_array because float32(pi/2) > float64(pi/2)
+    and simtel using float32.
+
+    Astropy complains about these values, so we fix them here.
+    """
+    if altitude > _half_pi and altitude < _half_pi_maxval:
```
Comment (nbiederbeck) [inline ctapipe/io/simteleventsource.py:89]:
```python 
if _hal_pi < altitude < _half_pi_maxval:
    return _half_pi
```

**Review by kosack — DISMISSED** [review]

File: ctapipe/io/simteleventsource.py (hunk 2 above)
Comment (maxnoe) [reply ctapipe/io/simteleventsource.py:89]: this is *very* slightly slower than the current code.

File: ctapipe/io/simteleventsource.py (hunk 1 above)
Comment (maxnoe) [reply ctapipe/io/simteleventsource.py]: fixed in the docstring of the new method

**Review by nbiederbeck — APPROVED** [review]

**Comment by maxnoe** [comment]
Fixed, please review again @LukasNickel @nbiederbeck

**Review by nbiederbeck — APPROVED** [review]

**Review by LukasNickel — APPROVED** [review]

**Pull request merged**